purchase_order_move_request/models/purchase_order.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    move_request_ids = fields.One2many(
+        comodel_name='account.move.request',
+        inverse_name='purchase_id',
+        string='Move Requests',
+    )
+    move_reaquest_count = fields.Integer(
+        string='Move Request Count',
+        compute='_compute_move_request_count',
+    )
+    @api.depends('move_request_ids')
+    def _compute_move_request_count(self):
+        for order in self:
+            order.move_reaquest_count = len(order.move_request_ids)
+
+    def action_move_request(self):
+        self.ensure_one()
+
+        line_vals = []
+        for line in self.order_line:
+            account = (
+                line.product_id.property_account_expense_id
+                or line.product_id.categ_id.property_account_expense_categ_id
+            )
+            line_vals.append(
+                Command.create(
+                    {
+                        "product_id": line.product_id.id,
+                        "name": line.name,
+                        "quantity": line.product_qty,
+                        "price_unit": line.price_unit,
+                        "account_id": account.id,
+                        "tax_ids": [Command.set(line.taxes_id.ids)],
+                        "analytic_distribution": line.analytic_distribution,
+                    }
+                )
+            )
```

Comment:
ต้อง refactor ให้ extract method ออกไปเป็น

_prepare_move_request_line_vals สำหรับ 

```suggestion
            line_vals.append(
                Command.create(line._prepare_move_request_line_vals())
            )
```

```py
class PurchaseOrderLine
	...

	def _prepare_move_request_line_vals(self):
		account = (
			self.product_id.property_account_expense_id
			or self.product_id.categ_id.property_account_expense_categ_id
		)
		return {
			"product_id": self.product_id.id,
            "name": self.name,
            "quantity": self.product_qty,
            "price_unit": self.price_unit,
            "account_id": account.id,
            "tax_ids": [Command.set(self.taxes_id.ids)],
            "analytic_distribution": self.analytic_distribution,
		}
```

Reply:
ขอแก้เป็น
```
def _prepare_move_request_line_vals(self , line):
		account = (
			line.product_id.property_account_expense_id
			or line.product_id.categ_id.property_account_expense_categ_id
		)
		return {
			"product_id": self.product_id.id,
            "name": line.name,
            "quantity": line.product_qty,
            "price_unit": line.price_unit,
            "account_id": account.id,
            "tax_ids": [Command.set(line.taxes_id.ids)],
            "analytic_distribution": line.analytic_distribution,
		}
```
เนื่องจากจะทำการส่งไลน์เข้ามาทำงานแทน